addons/trenches/functions/fnc_placeTrench.sqf

```diff
@@ -19,44 +19,48 @@
 params ["_unit", "_trenchClass"];
 
 //Load trench data
-private _noGeoModel = getText (configFile >> "CfgVehicles" >> _trenchClass >> QGVAR(noGeoClass));
-if(_noGeoModel == "") then {_noGeoModel = _trenchClass;};
-
-GVAR(trenchClass) = _trenchClass;
-GVAR(trenchPlacementData) = getArray (configFile >> "CfgVehicles" >> _trenchClass >> QGVAR(placementData));
+GVAR(trenchPlacementData) = getArray (configFile >> "CfgVehicles" >> _trenchClass >> QGVAR(PlacementData));
 TRACE_1("",GVAR(trenchPlacementData));
 
 // prevent the placing unit from running
-[_unit, "forceWalk", "ACE_Trenches", true] call EFUNC(common,statusEffect_set);
-[_unit, "blockThrow", "ACE_Trenches", true] call EFUNC(common,statusEffect_set);
+[_unit, "forceWalk", "ACE_Trenches", true] call ace_common_fnc_statusEffect_set;
 
 // create the trench
-private _trench = createVehicle [_noGeoModel, [0, 0, 0], [], 0, "NONE"];
-
+private _trench = createSimpleObject [_trenchClass, [0, 0, 0]];
 GVAR(trench) = _trench;
 
 // prevent collisions with trench
 [QEGVAR(common,enableSimulationGlobal), [_trench, false]] call CBA_fnc_serverEvent;
 
 GVAR(digDirection) = 0;
+GVAR(currentSurface) = "";
 
 // pfh that runs while the dig is in progress
 GVAR(digPFH) = [{
     (_this select 0) params ["_unit", "_trench"];
 
     // Cancel if the helper object is gone
     if (isNull _trench) exitWith {
-        [_unit] call FUNC(placeCancel);
+        [_unit] call ace_trenches_fnc_placeCancel;
     };
 
     // Cancel if the place is no longer suitable
-    if !([_unit] call FUNC(canDigTrench)) exitWith {
-        [_unit] call FUNC(placeCancel);
+    private _checkVar = [_unit] call ace_trenches_fnc_canDigTrench;
+    if ((typeName _checkVar) == "Number") then {
+      if (_checkVar > 0) then {
```

Comment:
if true then true else false? WAT
`_checkVar = _checkVar > 0;`
Also missing spaces on the else.

Reply:
I'm just converting numbers to bool here as the passed variable can be a number or a bool

Reply:
>if true then true else false

Is always dumb and bullshit.
Use what I wrote.